js/src/components/google-combo-account-card/connected-google-combo-account-card.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import AccountCard, { APPEARANCE } from '../account-card';
+import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
+import useGoogleMCAccount from '.~/hooks/useGoogleMCAccount';
+import AppSpinner from '../app-spinner';
+import useAutoCreateAdsMCAccounts from '../../hooks/useAutoCreateAdsMCAccounts';
+import LoadingLabel from '../loading-label/loading-label';
+import AccountCreationDescription from './account-creation-description';
+
+/**
+ * Clicking on the "connect to a different Google account" button.
+ *
+ * @event gla_google_account_connect_different_account_button_click
+ */
+
+/**
+ * Renders a Google account card UI with connected account information.
+ * It will also kickoff Ads and Merchant Center account creation if the user does not have accounts.
+ *
+ * @param {Object} props React props.
+ * @param {{ googleAccount: object }} props.googleAccount The Google account.
+ *
+ * @fires gla_google_account_connect_different_account_button_click
+ */
+const ConnectedGoogleComboAccountCard = ( { googleAccount } ) => {
+	const {
+		googleMCAccount,
+		hasFinishedResolution: hasFinishedResolutionForCurrentMCAccount,
+	} = useGoogleMCAccount();
+
+	const {
+		googleAdsAccount,
+		hasFinishedResolution: hasFinishedResolutionForCurrentAdsAccount,
+	} = useGoogleAdsAccount();
+
+	const {
+		accountsCreated,
+		accountCreationChecksResolved,
+		isCreatingAdsAccount,
+		isCreatingMCAccount,
+	} = useAutoCreateAdsMCAccounts();
+
+	const isCreatingAccounts = isCreatingAdsAccount || isCreatingMCAccount;
+
+	if (
+		! accountsCreated &&
+		( ! hasFinishedResolutionForCurrentAdsAccount ||
```

Comment:
No. This is required because these resolution checks will run twice.

1. When the Google account is connected (card is mounted)
2. When the accounts have been created.

If we remove these checks, the UI state will change between Creating accounts and then showing account.